internal/database/mysql.go

```diff
@@ -88,25 +86,25 @@ func (db *MySQLDatabase) Close() (err error) {
 	return
 }
 
-func (db *MySQLDatabase) AuthenticateLogin(username string, password string) (id int, err error) {
+func (db *MySQLDatabase) AuthenticateLogin(email, password string) (id uint64, err error) {
 	var hash []byte
-	err = db.underlyingDB.QueryRow("SELECT id, password FROM users WHERE BINARY username = ?", username).Scan(
+	err = db.underlyingDB.QueryRow("SELECT id, password FROM users WHERE BINARY email = ?", email).Scan(
 		&id, &hash,
 	)
 	if errors.Is(err, sql.ErrNoRows) {
-		return 0, errors.New("invalid username")
+		return 0, errors.New("invalid email")
 	}
 	if err = bcrypt.CompareHashAndPassword(hash, []byte(password)); err != nil {
 		return 0, errors.New("invalid password")
 	}
 	return
 }
 
-func (db *MySQLDatabase) GetUser(id int) (user *internal.User) {
+func (db *MySQLDatabase) GetUser(id uint64) (user *internal.User) {
 	user = &internal.User{}
 	err := db.underlyingDB.QueryRow("SELECT * FROM users WHERE id = ?", id).Scan(
-		&user.ID, &user.Username, &user.Email, &user.Password, &user.CreatedOn,
-	)
+		&user.ID, &user.Firstname, &user.Middlename, &user.Lastname, &user.Email, &user.Password, &user.DateofBirth,
```

Comment:
Do: FirstName, &user.MiddleName, &user.LastName

camel case capitalize "XyzName" since we are using underscores now.

Reply:
Changed